src/java.management/share/classes/javax/management/loading/MLetParser.java

```diff
@@ -249,7 +251,7 @@ public List<MLetContent> parse(URL url) throws IOException {
     /**
      * Parse the document pointed by the URL urlname
      */
-    @SuppressWarnings("deprecation")
+    @SuppressWarnings({"deprecation", "removal"})
```

Comment:
You might have to keep "removal", but I suspect "deprecation" should no longer be necessary, now that this class is deprecated (though I might be wrong).

Reply:
OK this took me a few rebuilds to figure out:

Before these changes, MLetParser needs the deprecation suppressions, for URL.  If you remove that suppression, and you get warnings.

When I make MLetParser itself deprecated for removal, it no longer needs the deprecation suppression for URL, but does need suppression of e.g. MLet which is tagged for removal.

i.e. in a class marked for removal, we warn you for using other classes marked for removal, but not for using others only deprecated (as presumably the removal will happen before the simply deprecated class is removed).

So yes we can remove "deprecation" from that line, and only have "removal".  It's fine either way.  I can take it out as you've mentioned it.